src/v/cluster/rm_group_frontend.cc

```diff
@@ -45,4 +45,401 @@ rm_group_frontend::rm_group_frontend(
       config::shard_local_cfg().metadata_dissemination_retry_delay_ms.value()) {
 }
 
+ss::future<begin_group_tx_reply> rm_group_frontend::begin_group_tx(
+  kafka::group_id group_id,
+  model::producer_identity pid,
+  model::timeout_clock::duration timeout) {
+    auto ntp_opt = _coordinator_mapper.local().ntp_for(group_id);
+    if (!ntp_opt) {
+        vlog(
+          clusterlog.trace,
+          "can't find ntp for {}, creating a consumer group topic",
+          group_id);
+        auto has_created = co_await try_create_consumer_group_topic();
+        if (!has_created) {
+            vlog(
+              clusterlog.warn, "can't create consumer group topic", group_id);
+            co_return begin_group_tx_reply{.ec = tx_errc::partition_not_exists};
+        }
+    }
+
+    auto retries = _metadata_dissemination_retries;
+    auto delay_ms = _metadata_dissemination_retry_delay_ms;
+    std::optional<model::node_id> leader_opt = std::nullopt;
+    tx_errc ec;
+    while (!leader_opt && 0 < retries--) {
+        ntp_opt = _coordinator_mapper.local().ntp_for(group_id);
+        if (!ntp_opt) {
+            vlog(clusterlog.trace, "can't find ntp for {}, retrying", group_id);
+            ec = tx_errc::partition_not_exists;
+            co_await ss::sleep(delay_ms);
+            continue;
+        }
```

Comment:
is this needed? it looks like `ntp_opt` will always have a value given what happens above at the beginning of the method.

Reply:
not really `try_create_consumer_group_topic` fills `_coordinator_mapper` but it doesn't happen instantly so there is a lag after it yields control flow but before `_coordinator_mapper` is populated